scripts/pylib/twister/twisterlib.py

```diff
@@ -3830,11 +3830,13 @@ def json_report(self, filename, append=False, version="NA"):
                     if rom_size:
                         testcase["rom_size"] = rom_size
 
-                    if instance.results[k] in ["PASS"] or instance.status == 'passed':
+                    if instance.results[k] in ["SKIP"] or instance.status == 'skipped':
+                        testcase["status"] = "skipped"
+                        testcase["reason"] = instance.reason
+                    elif instance.results[k] in ["PASS"] or instance.status == 'passed':
```

Comment:
so would this be betrter?
```suggestion
                    elif instance.results[k] in ["PASS"] and instance.status == 'passed':
```